src/worldcereal/openeo/inference.py

```diff
@@ -42,18 +42,18 @@ def predict(self, features: np.ndarray) -> np.ndarray:
         # Prepare input data for ONNX model
         outputs = self.onnx_session.run(None, {"features": features})
 
-        # Threshold for binary conversion
-        threshold = 0.5
+        # Get the prediction labels
+        binary_labels = (outputs[0] == "True").astype(np.uint8).reshape((-1, 1))
 
-        # Extract all prediction values and convert them to binary labels
-        prediction_values = np.array([sublist["True"] for sublist in outputs[1]])
-        binary_labels = prediction_values >= threshold
-        binary_labels = binary_labels.astype("uint8")
-
-        prediction_values = prediction_values * 100.0
-        prediction_values = np.round(prediction_values).astype("uint8")
-
-        return np.stack([binary_labels, prediction_values], axis=0)
+        # Extract all probabilities
+        all_probabilities = np.round(
+            np.array([[x["False"], x["True"]] for x in outputs[1]]) * 100.0
+        ).astype(np.uint8)
+        max_probability = np.max(all_probabilities, axis=1, keepdims=True)
+        # return np.concatenate(
```

Comment:
i don't really understand. This was temporary version of me. We're now not returning all probabilities, so how can we smooth downstream?

Reply:
Here we are in cropland classification, which is binary classification so you only need one probability to know the other. There is a parameter for the postprocessor called "is_binary" that takes this into account.

Reply:
ah ok that's once way. Not sure if that's easier to understand than just implementing the same functionality in a binary classifier instead of dealing with it in a postprocessor. But I guess ok for now. Should we remove the commented code here then? Some of the code I wrote is now actually convoluted if individual probabilities are not used.